### PR TITLE
FS-927: add mtcl cmd for uploadbom API

### PR DESCRIPTION
**Pull request by Alva8756**
This PR add mtcl cmd for uploadbom API

Tested with sandbox server service <--> local bomservice <---> mctl client

### CODE REVIEW

**Review by jakeschuurmans — APPROVED** [review]

File: internal/app/clients.go [hunk 1]
```diff
@@ -64,3 +65,25 @@ func NewConditionsClient(ctx context.Context, cfg *model.ConfigOIDC, reauth bool
...
+		return nil, errors.Wrap(ErrNilConfig, "missing bom service API client configuration")
+	}
+
+	if cfg.Disable {
+		return bomclient.NewClient(
+			cfg.Endpoint,
+		)
+	}
+
+	token, err := auth.AccessToken(ctx, model.BomsServiceAPI, cfg, reauth)
```
Comment (DoctorVin) [inline internal/app/clients.go:80]: This code looks fine, but there is some setup we'll have to do to get BOMService to get things to authenticate properly, unless we're using the same Okta groups and stuff for BOMService as we do for ConditionOrc (and that would probably be fine).

File: cmd/create/bomupload.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	// Xlsx file containing one or multiple boms information.
+	bomXlsxFile string
+}
+
+var (
+	flagsUploadBomFileFlags *uploadBomFileFlags
+)
+
+var uploadBomFile = &cobra.Command{
+	Use:   "uploadbom",
```
Comment (DoctorVin) [inline cmd/create/bomupload.go:23]: This has a weird sound in English to me. What about `create bom` and (like you do), call out specifically in the help that this is uploading a BOM to BOMService?

**Review by DoctorVin — APPROVED** [review]

File: cmd/create/bomupload.go (hunk 2 above)
Comment (Alva8756) [reply cmd/create/bomupload.go:23]: Thanks! Will include it in the the follow-up PR.

File: internal/app/clients.go (hunk 1 above)
Comment (Alva8756) [reply internal/app/clients.go:80]: Thanks for pointing it out.

I am not familiar with okta groups. Currently I found oidc client id `0oaa8kto5lZYascqP5d7`, does it matches a group in okta? Can you help provide me a link to the okta configuration that I can learn and play around? In the future we may need to have a separate group for bom service.

Thanks!

**Pull request merged**